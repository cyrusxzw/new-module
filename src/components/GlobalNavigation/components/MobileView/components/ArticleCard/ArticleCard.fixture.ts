import type { ArticleCardProps } from './ArticleCard.types';

const ArticleCardFixture: ArticleCardProps = {
  id: 'article',
  label: 'Article Card Label',
  title: 'See all Skin Care Card',
  url: '#article',
  metaLabel: 'Article Card  â€¢  Meta Label',
  image: {
    altText: 'Image of Article Card',
    sizes: {
      large:
        'https://www.aesop.com/medias/Aesop-Room-Sprays-Hybris-Cythera-Pattern-Desktop-1440x1600.jpg?context=bWFzdGVyfGltYWdlc3w0MTU4Mjl8aW1hZ2UvanBlZ3xpbWFnZXMvaGIyL2g0MS84ODEzNzIxNTUwODc4LmpwZ3wxODAxZWQ5YTdlNzMxNDYxZWZhMThiNzQ1YTJhZTRmMjE3ZWU3YzEzZDJmOGZjZWU0M2Y4ZWJlNjg2YjE1NjQ2',
      medium:
        'https://www.aesop.com/medias/Aesop-Room-Sprays-Hybris-Cythera-Pattern-Desktop-1440x1600.jpg?context=bWFzdGVyfGltYWdlc3w0MTU4Mjl8aW1hZ2UvanBlZ3xpbWFnZXMvaGIyL2g0MS84ODEzNzIxNTUwODc4LmpwZ3wxODAxZWQ5YTdlNzMxNDYxZWZhMThiNzQ1YTJhZTRmMjE3ZWU3YzEzZDJmOGZjZWU0M2Y4ZWJlNjg2YjE1NjQ2',
      small:
        'https://www.aesop.com/medias/Aesop-Room-Sprays-Hybris-Cythera-Pattern-Desktop-1440x1600.jpg?context=bWFzdGVyfGltYWdlc3w0MTU4Mjl8aW1hZ2UvanBlZ3xpbWFnZXMvaGIyL2g0MS84ODEzNzIxNTUwODc4LmpwZ3wxODAxZWQ5YTdlNzMxNDYxZWZhMThiNzQ1YTJhZTRmMjE3ZWU3YzEzZDJmOGZjZWU0M2Y4ZWJlNjg2YjE1NjQ2',
    },
  },
};

export { ArticleCardFixture };
