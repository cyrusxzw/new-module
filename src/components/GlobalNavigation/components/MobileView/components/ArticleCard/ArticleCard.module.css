@import 'styles/index';

.base {
  min-width: 80vw;
  box-sizing: border-box;
  padding-top: 16px;
  padding-bottom: 16px;
  scroll-snap-align: center;

  &.dark {
    color: var(--color-dark-label);
  }

  &.light {
    color: var(--color-light-label);
  }
}

.image {
  overflow: hidden;
  max-height: 240px;
  aspect-ratio: 16 / 9;
  pointer-events: none;

  img {
    width: 100%;
    height: auto;
    aspect-ratio: 16 / 9;
    object-fit: cover;
  }
}

.metaLabel {
  margin-top: 12px;
  font-family: var(--font-family-sans-serif-regular);
  line-height: 1.4;
}

.label {
  margin-bottom: 0;
  font-size: rem(20px);
  line-height: 1.33;
}

.leftAligned {
  flex-direction: column;
  align-items: start;
}
