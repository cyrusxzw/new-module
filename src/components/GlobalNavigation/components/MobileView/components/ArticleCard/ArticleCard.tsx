import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import { Hyperlink } from '~/components/Hyperlink';
import { Paragraph } from '~/components/Paragraph';
import { Image } from '~/components/Image';
import type { ArticleCardType } from './ArticleCard.types';
import compositionStyles from '../../MobileView.module.css';
import styles from './ArticleCard.module.css';

const ArticleCard: ArticleCardType = ({
  image,
  isVisible = true,
  label,
  metaLabel,
  title,
  url,
}) => {
  const currentTheme = useThemeContext(null, 'dark');

  const classSet = cx(
    styles.base,
    styles[currentTheme],
    compositionStyles.itemElement,
    compositionStyles.ornamentalWrapper,
    styles.leftAligned,
    styles.mobileArticle,
  );

  return (
    <Hyperlink
      className={classSet}
      tabIndex={!isVisible ? -1 : null}
      theme={currentTheme}
      title={title}
      url={url}
    >
      <Image {...image} className={styles.image} />

      {metaLabel && (
        <Paragraph
          className={styles.metaLabel}
          isFlush={true}
          theme={currentTheme}
        >
          {metaLabel}
        </Paragraph>
      )}

      <Paragraph
        className={styles.label}
        hasSerifFont={true}
        isFlush={true}
        theme={currentTheme}
      >
        <span className={compositionStyles.ornamentalHover}>{label}</span>
      </Paragraph>
    </Hyperlink>
  );
};

export { ArticleCard };
