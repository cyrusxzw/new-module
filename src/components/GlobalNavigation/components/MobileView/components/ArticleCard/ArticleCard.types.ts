import type { ComponentWithoutChildren } from '~/types';
import type { Clickable } from '~/components/GlobalNavigation/GlobalNavigation.types';

type ArticleCardProps = Clickable & {
  isVisible?: boolean;
  metaLabel?: string;
  url: string;
  image?: {
    altText: string;
    sizes: {
      large?: string;
      medium?: string;
      small?: string;
      xLarge?: string;
      xSmall?: string;
    };
  };
};

type ArticleCardType = ComponentWithoutChildren<ArticleCardProps>;

export type { ArticleCardType, ArticleCardProps };
