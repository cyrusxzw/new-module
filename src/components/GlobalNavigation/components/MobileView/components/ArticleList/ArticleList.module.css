@import 'styles/index';

.base {
  margin-bottom: 40px;
  opacity: 1;
  transition: opacity 200ms var(--easing-ease-out-cubic);
}

.articleScroll {
  display: -ms-flexbox;
  display: flex;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  overflow-x: auto;
  -ms-scroll-snap-type: x mandatory;
  scroll-snap-type: x mandatory;
}

.hidden {
  opacity: 0;
  pointer-events: none;
}
