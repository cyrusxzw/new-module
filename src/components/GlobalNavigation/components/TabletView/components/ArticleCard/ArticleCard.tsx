import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import { Hyperlink } from '~/components/Hyperlink';
import { Image } from '~/components/Image';
import { Paragraph } from '~/components/Paragraph';
import type { ArticleCardType } from './ArticleCard.types';
import compositionStyles from '../../TabletView.module.css';
import styles from './ArticleCard.module.css';

const ArticleCard: ArticleCardType = ({
  image,
  isVisible = true,
  label,
  metaLabel,
  title,
  url,
}) => {
  const currentTheme = useThemeContext(null, 'dark');

  const classSet = cx(
    styles.base,
    styles[currentTheme],
    compositionStyles.ornamentalWrapper,
    styles.leftAligned,
  );

  return (
    <Hyperlink
      className={classSet}
      tabIndex={!isVisible ? -1 : null}
      theme={currentTheme}
      title={title}
      url={url}
    >
      {metaLabel && (
        <Paragraph
          className={styles.metaLabel}
          isFlush={true}
          theme={currentTheme}
        >
          {metaLabel}
        </Paragraph>
      )}

      <Paragraph
        className={styles.label}
        hasSerifFont={true}
        isFlush={true}
        theme={currentTheme}
      >
        <span className={compositionStyles.ornamentalHover}>{label}</span>
      </Paragraph>

      <Image {...image} className={styles.image} height="195" />
    </Hyperlink>
  );
};

export { ArticleCard };
