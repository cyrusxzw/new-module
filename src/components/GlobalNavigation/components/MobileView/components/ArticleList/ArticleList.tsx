import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import { useGlobalNavigationStateContext } from '~/components/GlobalNavigation/GlobalNavigation.context';
import { ArticleCard } from '../ArticleCard';
import type { ArticleListType } from './ArticleList.types';
import styles from './ArticleList.module.css';

const ArticleList: ArticleListType = ({ className, theme, items }) => {
  const { isOpen, activeCollectionId } = useGlobalNavigationStateContext();
  const currentTheme = useThemeContext(theme, 'dark');

  if (!items?.length) return null;

  const isVisible = isOpen && activeCollectionId === 'top';

  const articleScrollSet = styles.articleScroll;

  const classSet = cx(
    styles.base,
    { [styles.hidden]: activeCollectionId !== 'top' },
    styles[currentTheme],
    className,
  );

  return (
    <div className={classSet}>
      <div className={styles.cardHeading}>The Athenaeum</div>
      <aside aria-label="navigation read articles" className={articleScrollSet}>
        {items?.map((item) => (
          <ArticleCard {...item} isVisible={isVisible} key={item.id} />
        ))}
      </aside>
    </div>
  );
};

export { ArticleList };
