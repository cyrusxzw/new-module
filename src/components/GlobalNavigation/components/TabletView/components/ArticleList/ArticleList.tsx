import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import { ArticleCard } from '../ArticleCard';
import type { ArticleListType } from './ArticleList.types';
import styles from './ArticleList.module.css';

const ArticleList: ArticleListType = ({
  className,
  isVisible,
  items,
  theme,
}) => {
  const currentTheme = useThemeContext(theme, 'dark');

  if (!items?.length) return null;

  const classSet = cx(
    styles.base,
    { [styles.hidden]: !isVisible },
    styles[currentTheme],
    className,
  );

  return (
    <div className={classSet}>
      <aside aria-label="navigation read articles">
        {items.map((item) => (
          <ArticleCard {...item} isVisible={isVisible} key={item.id} />
        ))}
      </aside>
    </div>
  );
};

export { ArticleList };
