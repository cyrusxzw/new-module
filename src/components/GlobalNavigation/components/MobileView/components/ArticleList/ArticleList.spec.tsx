/* eslint-disable init-declarations */
import React from 'react';
import { render } from '@testing-library/react';
import { ArticleList } from './ArticleList';
import { ArticleListFixture } from './ArticleList.fixture';
import * as globalNavigationContextFile from '~/components/GlobalNavigation/GlobalNavigation.context';
import * as articleCardFile from '../ArticleCard/ArticleCard';

describe('<ArticleList />', () => {
  let useGlobalNavigationStateContextSpy: jest.SpyInstance;
  let articleCardSpy: jest.SpyInstance;

  const TestBed = ({ items }) => (
    <globalNavigationContextFile.GlobalNavigationStateContextProvider>
      <ArticleList items={items} />
    </globalNavigationContextFile.GlobalNavigationStateContextProvider>
  );

  beforeEach(() => {
    jest.clearAllMocks();
    useGlobalNavigationStateContextSpy = jest
      .spyOn(globalNavigationContextFile, 'useGlobalNavigationStateContext')
      .mockReturnValue({ isOpen: false, activeCollectionId: 'top' } as any);
    articleCardSpy = jest
      .spyOn(articleCardFile, 'ArticleCard')
      .mockImplementation(() => null);
  });

  it('should render base component correctly', () => {
    const { container } = render(<TestBed items={undefined} />);

    expect(container).toMatchSnapshot();
  });

  it('should render with items', () => {
    render(<TestBed items={ArticleListFixture.items} />);

    expect(articleCardSpy).toHaveBeenCalledTimes(
      ArticleListFixture.items.length,
    );
  });

  it('should render with items and show cards', () => {
    useGlobalNavigationStateContextSpy.mockReturnValue({
      isOpen: true,
      activeCollectionId: 'top',
    });

    render(<TestBed items={ArticleListFixture.items} />);

    expect(articleCardSpy).toHaveBeenCalledWith(
      expect.objectContaining({ isVisible: true }), // Props
      expect.anything(), // Context
    );
  });
});
