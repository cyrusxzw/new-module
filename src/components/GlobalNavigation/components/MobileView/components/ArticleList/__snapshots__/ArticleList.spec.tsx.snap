// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleList /> should render base component correctly 1`] = `<div />`;
